### ISSUE TITLE
Any chance this will work with an ISY99i?

### ISSUE THREAD

**Issue by ModMike** [bgottsch/homebridge-isy-hub#2]

**Comment by bgottsch** [comment]
Hey @ModMike ,

Probably yes, I don't own one, so I haven't tested it. As I understand from Universal Devices, all of the ISY hubs support REST, which is what is used by this plugin. It also uses a WebSocket, which is part of the REST API, so no problem here. Therefore, as said, not tested.

I have recently rebuilt this plugin and have some thing to finish, such as the README and stateless switches, but it's working in it's current state. npm version should be 0.0.4

Take a look at the file config.example.json to see how to configure it.

I should be done with this plugin very shortly, I just have some work stuff to prioritize first.

If you have any questions, or if an error appears, let me know so I can help.

Best regards

**Comment by ModMike** [comment]
But do you need version 5 of the firmware to make it work? ISY99i stoped at version 3.1.10 or something like that.

**Comment by bgottsch** [comment]
Let me check in the Universal Devices website for the info.

It might be simpler to just try this plugin and to see if it works. The ip should be ISY ip in your LAN. Username and password should be the same as used to access the portal.

(edited)
If you use homebridge-config-ui-x, this plugin is avaliable there and makes this test much easier.

**Comment by ModMike** [comment]
Ok will try and let you know.

**Comment by bgottsch** [comment]
From their website:

If you search for ISY99i REST API, this page pops up: `https://wiki.universal-devices.com/index.php?title=ISY-99i_Series_INSTEON:REST_Interface&redirect=no`

That page has a redirect link to the same API used by the ISY994i

Hope that's the case

**Comment by ModMike** [comment]
Ok JSON ready, do I change platform name or does that even matter?

**Comment by bgottsch** [comment]
Platform name has to be fixed (isy-hub) and name can be anything (defaults to isy-hub), it does not matter (but it's still required). This would affect the log outputs on homebridge

**Comment by ModMike** [comment]
Ok changed to isy hub but get this error: 

[homebridge-isy-hub.isy-hub] Disabling plugin, wrong config declaration or not found!

**Comment by bgottsch** [comment]
Can you share the config file? (Ommit devices and scenes and senstive info)

**Comment by bgottsch** [comment]
Ahh, another think poped to my mind, how did you install it? Since the instructions on the README are incorrect. The correct install should be `npm install -g homebridge-isy-hub`

**Comment by ModMike** [comment]
When I load my iso, it says the name is My. Could that be the issue? How can I confirm the name?

I installed using the plug in finder, I think. Hard to remember, I just set it up.

**Comment by ModMike** [comment]
I put Xs for all sensitive info:

```
{
    "bridge": {
        "name": "Homebridge XXXX",
        "username": "XX:XX:XX:XX:XX",
        "port": XXXXXX,
        "pin": "XXX-XX-XXX"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": XXXX,
            "platform": "config"
        }
    ],
    "platform": "isy-hub",
    "name": "isy-hub",
    "username": "XXXXX",
    "password": "XXXXX",
    "host": "XXXXX",
    "refresh_interval": 30000,
    "scenes": [
        {
            "name": "Light Scene",
            "id": 12345,
            "type": "light_scene"
        },
        {
            "name": "Toggle Scene",
            "id": 23456,
            "type": "toggle_scene",
            "on_level": 76
        },
        {
            "name": "Window Scene",
            "id": 34567,
            "type": "window_scene"
        }
    ]
}
```

**Comment by bgottsch** [comment]
The problem is with the config file. Probably you used the one from the README as basis. Use the one from the file `config.example.json`

Here's a quick edit of your file:

```
{
	"bridge": {
		"name": "Homebridge XXXX",
		"username": "XX:XX:XX:XX:XX",
		"port": XXXXXX,
		"pin": "XXX-XX-XXX"
	},
	"accessories": [],
	"platforms": [
		{
			"name": "Config",
			"port": XXXX,
			"platform": "config"
		},
		{
			"platform": "isy-hub",
			"name": "isy-hub",
			"username": "XXXXX",
			"password": "XXXXX",
			"host": "XXXXX",
			"refresh_interval": 30000,
			"devices": [],
			"scenes": [
				{
					"name": "Light Scene",
					"id": 12345,
					"type": "lightbulb",
					"on_level": 100
				},
				{
					"name": "Toggle Scene",
					"id": 23456,
					"type": "switch",
					"on_level": 76
				},
				{
					"name": "Window Scene",
					"id": 34567,
					"type": "window_cover"
				}
			]
		}
	]
}
```

**Comment by ModMike** [comment]
Thank you very much for taking the time to do this for me. I gave it a shot this morning but I got the errors below. Am I supposed to define each device and scene manually?

[4/20/2020, 10:28:27] [isy-hub] Socket unreachable, attempting reconnect...
[4/20/2020, 10:28:32] [isy-hub] Accessory unreachable (name: Light Scene, id: 12345, class: scene, type: lightbulb)
[4/20/2020, 10:28:32] [isy-hub] Accessory unreachable (name: Toggle Scene, id: 23456, class: scene, type: switch)
[4/20/2020, 10:28:32] [isy-hub] Accessory unreachable (name: Window Scene, id: 34567, class: scene, type: window_cover)
[4/20/2020, 10:28:32] [isy-hub] Socket unreachable, attempting reconnect…

**Comment by bgottsch** [comment]
Hey @ModMike 

Great to know config is working. Yes, unfortunatelly, I designed this plugin with manual adding of devices and scenes. This is due to a design choice of manually chosing the accessory type in the config file, rather than making the plugin automatically choose it based on the device/scene. And it would be pointless to auto add and create a mapping in the config.

Anyway, about your errors: the plugin tries to connect to the ISY web scoket, and if not possbile, sets the reachability to false. Thus, your problem is with not being able to connect to the web socket.

Let's investigate, if you're up to it. On the ISY admin, click on the menu Tools > Diagnostics > Event Viewer. A windows should pop up, this is the output of the web socket, check if anything logs if you switch a device.

**Comment by ModMike** [comment]
Absolutely up to it, will look at it during lunch.

**Comment by bgottsch** [comment]
> But do you need version 5 of the firmware to make it work? ISY99i stoped at version 3.1.10 or something like that.

From ISY Portal: `Starting with firmware version 4.2.3, you can now use REST to subscribe to ISY using Web Sockets.`

Unfortunately, it might be the case where the ISY-99i won't work with this plugin. Still, it supports REST, but not web sockets. It might be possible to create a plugin without web sockets, but i'd have to change a lot of my logic for it.

Take a look at this other ISY [plugin](https://github.com/pradeepmouli/homebridge-isy-js). Don't know if it will work for the ISY-99i, but might be worth the try.

(Ps: opinion about buying an ISY-994i in 2020)

I purchased my Insteon home automation setup a while back, when home automation and specially lighting control was in it's early days. During my experience, integration was the most consuming task, since HomeKit had very strict design limitations, and Insteon also had theirs. ISY-994i was the second hub I have tried to use for my Insteon setup, and it was mostly worked great.

Therefore, there were some hickups on the way - scene dimming, grouping accessories and controlling via REST with no lag, controlling KeyPads, and the list goes on. Therefore, I have decided it's now time to revise this setup, and maybe start to improve it.

After an extensive research, I'm now experimenting with ESC8266 wi-fi controllers and other types of dimmers. This topic is complex, and I haven't had much progress yet, but with this, it's possible to use a $5 Sonoff device, and flash it to have native HomeKit support. This would solve most of the issues I currently have with Insteon, but it means replacing my system. If I get some good results from it, I might migrate little by little into the newer platforms.

I also came across the [Shelly Dimmer](https://shelly.cloud/wifi-smart-home-automation-shelly-dimmer/), a device similar to the Insteon Micro Dimmer, but with native wi-fi and MQTT support (ESP8266 built-in). They are relatively cheap and very compatible with most home automation systems. There's already a homebridge plugin to auto discover and add Shelly devices. Purchased mine yesterday, and will test more when it arrives.

In conclusion, is it worth it to buy an ISY-994i in 2020? Maybe, depends on your budget and necessity. Therefore, I believe it's time to invest into newer alternatives, such as Shelly (not tested yet), or others.

**Comment by ModMike** [comment]
Good point but I already invested in Insteon 7 years ago and have about 30 light switches and keypads. I think I will just bite the bullet and get it.

I don’t want to waste your time on outdated hardware.

Thanks!

**Comment by bgottsch** [comment]
No worries! Glad to help

**Comment by ModMike** [comment]
> > But do you need version 5 of the firmware to make it work? ISY99i stoped at version 3.1.10 or something like that.
> 
> From ISY Portal: `Starting with firmware version 4.2.3, you can now use REST to subscribe to ISY using Web Sockets.`
> 
> Unfortunately, it might be the case where the ISY-99i won't work with this plugin. Still, it supports REST, but not web sockets. It might be possible to create a plugin without web sockets, but i'd have to change a lot of my logic for it.
> 
> Take a look at this other ISY [plugin](https://github.com/pradeepmouli/homebridge-isy-js). Don't know if it will work for the ISY-99i, but might be worth the try.
> 
> (Ps: opinion about buying an ISY-994i in 2020)
> 
> I purchased my Insteon home automation setup a while back, when home automation and specially lighting control was in it's early days. During my experience, integration was the most consuming task, since HomeKit had very strict design limitations, and Insteon also had theirs. ISY-994i was the second hub I have tried to use for my Insteon setup, and it was mostly worked great.
> 
> Therefore, there were some hickups on the way - scene dimming, grouping accessories and controlling via REST with no lag, controlling KeyPads, and the list goes on. Therefore, I have decided it's now time to revise this setup, and maybe start to improve it.
> 
> After an extensive research, I'm now experimenting with ESC8266 wi-fi controllers and other types of dimmers. This topic is complex, and I haven't had much progress yet, but with this, it's possible to use a $5 Sonoff device, and flash it to have native HomeKit support. This would solve most of the issues I currently have with Insteon, but it means replacing my system. If I get some good results from it, I might migrate little by little into the newer platforms.
> 
> I also came across the [Shelly Dimmer](https://shelly.cloud/wifi-smart-home-automation-shelly-dimmer/), a device similar to the Insteon Micro Dimmer, but with native wi-fi and MQTT support (ESP8266 built-in). They are relatively cheap and very compatible with most home automation systems. There's already a homebridge plugin to auto discover and add Shelly devices. Purchased mine yesterday, and will test more when it arrives.
> 
> In conclusion, is it worth it to buy an ISY-994i in 2020? Maybe, depends on your budget and necessity. Therefore, I believe it's time to invest into newer alternatives, such as Shelly (not tested yet), or others.

I think that one of the things the OP is not considering is that the ISY994i and it's variants give you the kind of programatic control that you simply cannot have in Homekit. I am not seeing it is equal to control 4 in any way but it is kind of the same paradigm in the sense that you have powerful scripts and programs you can run.

Whats great is that can all be set up in the background and you can use HomeKit for manual control.

**Comment by bgottsch** [comment]
That’s true, ISY gives you more control, though there are better platforms such as HomeAssistant or even a simple MQTT server.

The ISY is terrific to control Insteon devices and a few others, and can easily be integrated with platforms as the ones mentioned above.

Therefore it’s still one more step than necessary. Home automation has evolved a lot, and IoT has mostly migrated from proprietary systems (such as Insteon) to more modern open-source systems. These allow for better overall integration, while not limiting features.

I base my opinion due to a series of frustrations while trying to simply control de brightness of multiple devices at the same time (not a scene, with preset values, but continuous control with HomeKit slider). This is a simple example where the Insteon protocol fails in comparison to others.

Nonetheless, Insteon has it’s merits and features, which are unique to it. But IMHO, there are better alternatives.

I saw an article back in Dec/19 stating that Insteon was experimenting with improvements and modernizations to their platform, let’s hope this happens.

**Comment by ModMike** [comment]
Got my 994i and everything works perfectly once I understood how to format the config file. Thanks for the help!

**Comment by bgottsch** [comment]
Great to hear that! Glad to help.

If you run into any trouble or have any suggestions, feel free to get in touch.

Best regards

**Comment by ModMike** [comment]
Your edit of my file was the key to understanding the JSON format and syntax. I do have once last remaining issue, but not sure if it's related: After the platform initializes I get this message:

```
[isy-hub] Startup complete!
(node:3960) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
```
Do I need to use https:// or the dhcp name of the ISY instead of IP to comply with SSL? Just guessing here.
